src/Bicep.Core/Semantics/Metadata/ResourceMetadataCache.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Concurrent;
+using Bicep.Core.Resources;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Semantics.Metadata
+{
+    public class ResourceMetadataParent
```

Comment:
>ResourceMetadataParent [](http://example.com/codeflow?start=17&length=22)

Could you move these to separate files? When/if we make them records we can put them back.